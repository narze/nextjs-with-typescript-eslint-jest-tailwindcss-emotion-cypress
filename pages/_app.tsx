import Head from 'next/head'
import '../styles/base.css'
import { ThemeProvider, theme } from '@chakra-ui/core'
import { GlobalStyles } from 'twin.macro'

export default function App({
  Component,
  pageProps,
}: {
  Component: any
  pageProps: any
}): JSX.Element {
  return (
    <>
      <Head>
        <title>
          Create Next App - with TypeScript, ESLint, Jest, TailwindCSS, Emotion
        </title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <GlobalStyles />
      <ThemeProvider theme={theme}>
        <Component {...pageProps} />
      </ThemeProvider>
    </>
  )
}
