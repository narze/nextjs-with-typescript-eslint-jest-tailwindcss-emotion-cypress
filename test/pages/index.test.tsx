import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../pages/index'
import { ThemeProvider, theme } from '@chakra-ui/core'

const renderWithProvider = (
  ui,
  { providerProps = {}, ...renderOptions } = {}
) => {
  return render(
    <ThemeProvider theme={theme} {...providerProps}>
      {ui}
    </ThemeProvider>,
    renderOptions
  )
}

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders with buttons alert', () => {
    const { getByText } = renderWithProvider(<Home />, {})

    getByText('Emotion CSS')
    getByText('Emotion React')
    getByText('Chakra-UI')
  })
})
