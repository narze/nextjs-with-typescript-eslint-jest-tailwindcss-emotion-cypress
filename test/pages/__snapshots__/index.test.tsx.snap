// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-2250570153 container"
  >
    <main
      class="jsx-2250570153"
    >
      <h1
        class="jsx-2250570153 title"
      >
        Welcome to 
        <a
          class="jsx-2250570153"
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="jsx-2250570153 css-1bdbpaa-subtitle"
        data-testid="subtitle"
      >
        with
        <span
          class="jsx-2250570153 css-19dihr3-highlight"
        >
          TypeScript
        </span>
        <span
          class="jsx-2250570153 css-19dihr3-highlight"
        >
          ESLint
        </span>
        <span
          class="jsx-2250570153 css-19dihr3-highlight"
        >
          Jest
        </span>
        <span
          class="jsx-2250570153 css-19dihr3-highlight"
        >
          TailwindCSS
        </span>
        <span
          class="jsx-2250570153 css-19dihr3-highlight"
        >
          Emotion
        </span>
        and
        <span
          class="jsx-2250570153 css-19dihr3-highlight"
        >
          Cypress
        </span>
      </p>
      <p
        class="jsx-2250570153 description"
      >
        Get started by editing 
        <code
          class="jsx-2250570153"
        >
          pages/index.tsx
        </code>
      </p>
      <button
        class="jsx-2250570153"
      >
        Test Button
      </button>
      <div
        class="jsx-2250570153 css-1d6h1jx-Home"
      >
        <button
          class="css-e7qmf6-button group"
        >
          <span
            class="css-1plhd0h-ButtonCss"
          >
            <svg
              class="css-qkfjq1-ButtonCss"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                clip-rule="evenodd"
                d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          @emotion/css
        </button>
        <button
          class="group  css-1tcz458-Component"
        >
          <span
            class="css-15quut2-ButtonReact"
          >
            <svg
              class="css-jxx9nt-ButtonReact"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                clip-rule="evenodd"
                d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          @emotion/react
        </button>
        <button
          class="group  css-xcz8wc-Button e1hljgas2"
        >
          <span
            class="css-1maqz9a-IconWrapper e1hljgas1"
          >
            <svg
              class="css-1yqsut3-Icon e1hljgas0"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                clip-rule="evenodd"
                d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          @emotion/styled
        </button>
      </div>
      <div
        class="jsx-2250570153 grid"
      >
        <a
          class="jsx-2250570153 card"
          href="https://nextjs.org/docs"
        >
          <h3
            class="jsx-2250570153"
          >
            Documentation →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="jsx-2250570153 card"
          href="https://nextjs.org/learn"
        >
          <h3
            class="jsx-2250570153"
          >
            Learn →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="jsx-2250570153 card"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h3
            class="jsx-2250570153"
          >
            Examples →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="jsx-2250570153 card"
          href="https://vercel.com/import?filter=next.js&utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h3
            class="jsx-2250570153"
          >
            Deploy →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="jsx-2250570153"
    >
      <a
        class="jsx-2250570153"
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <img
          alt="Vercel Logo"
          class="jsx-2250570153 logo"
          src="/vercel.svg"
        />
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
